### PR TITLE
Add Overlay Quota support

### PR DESCRIPTION
**Pull request by rhatdan**
This patch vendors in the latest containers/storage which include quota support for overlay storage on XFS file systems.

Also include updates to the crio.conf.5 man page to describe how to configure these quota options.

### CODE REVIEW

**Comment by rhatdan** [comment]
@mrunalp @nalind @rhvgoyal @runcom PTAL

And see if you can get quota setup in your environment.

We need to add tests for this, but the tests can only run on a specific storage setup, so we can not run them in general.

**Comment by rhatdan** [comment]
@cevich PTAL, also to see if you have ideas of how to do a test.

File: docs/crio.conf.5.md [hunk 1]
```diff
@@ -36,11 +36,29 @@ The `crio` table supports the following options:
...
 
 **storage_option**=[]
   CRIO storage driver option list (no default)
 
+  Values:
+
+	additionalimagestores=[]
+
+	Paths to additional container image stores. These are read/only and are usually stored on remote network shares.
+	storage_option=[ "overlay.additionaimagestores=/mnt/overlay" ]
```
Comment (nalind) [inline docs/crio.conf.5.md]: Should this be `overlay.imagestore=/mnt/overlay`?

Comment (rhatdan) [reply docs/crio.conf.5.md]: Yes fixed it.

**Comment by mrunalp** [comment]
Failures in test:
```
not ok 6 pids limit
# (from function `start_crio' in file ./helpers.bash, line 224,
#  in test file ./cgroups.bats, line 13)
#   `PIDS_LIMIT=1234 start_crio' failed
# 7:pids:/
# time="2017-09-26T18:06:42-04:00" level=error msg="error opening storage: overlay: the backing xfs filesystem is formatted without d_type support, which leads to incorrect behavior. Reformat the filesystem with ftype=1 to enable d_type support. Running without d_type is not supported." 
# rm: cannot remove ‘/tmp/tmp.WKtccoA3m4/crio/overlay2’: Device or resource busy
not ok 7 ctr not found correct error message
# (from function `start_crio' in file ./helpers.bash, line 224,
#  in test file ./ctr.bats, line 10)
#   `start_crio' failed
# time="2017-09-26T18:06:43-04:00" level=error msg="error opening storage: overlay: the backing xfs filesystem is formatted without d_type support, which leads to incorrect behavior. Reformat the filesystem with ftype=1 to enable d_type support. Running without d_type is not supported." 
# rm: cannot remove ‘/tmp/tmp.bp3NtaoHd7/crio/overlay2’: Device or resource busy
not ok 8 ctr termination reason Completed
````

**Comment by rhatdan** [comment]
@mrunalp So the question is whether we want to force users to run with d_type?  All freshly installed RHEL7.4 systems should have this enabled by default.  (I believe, @rhvgoyal?)  So can we up the tests machines to be RHEL7.4?  Or we can turn this error to a warning, but quota will not work on these systems.

**Comment by rhatdan** [comment]
@skuznets is looking into this

**Comment by stevekuznetsov** [comment]
bad code in the ami packaging step in the playbook I used, updated

/test ami

**Comment by mrunalp** [comment]
okay, so looked at https://ci.openshift.redhat.com/jenkins/job/test_pull_request_crio_ami_rhel/22/s3/ The crio.service failed to come up. However, this is expected because we would need the ami to boot up with 954 for the kernel cmdline flags to take effect. So, I think we will have to first get in 954 (drop that commit from here) and then this PR would be able to boot off ami created with 954 and pass the tests. 
```
-- Logs begin at Fri 2017-09-15 19:46:50 EDT, end at Thu 2017-09-28 20:06:13 EDT. --
Sep 28 20:06:02 ip-172-18-8-38.ec2.internal systemd[1]: Starting Open Container Initiative Daemon...
Sep 28 20:06:03 ip-172-18-8-38.ec2.internal crio[21492]: time="2017-09-28 20:06:03.055229394-04:00" level=warning msg="Using pre-4.0.0 kernel for overlay, mount failures may require kernel update"
Sep 28 20:06:03 ip-172-18-8-38.ec2.internal crio[21492]: time="2017-09-28 20:06:03.057001932-04:00" level=fatal msg="overlay: the backing xfs filesystem is formatted without d_type support, which leads to incorrect behavior. Reformat the filesystem with ftype=1 to enable d_type support. Running without d_type is not supported."
Sep 28 20:06:03 ip-172-18-8-38.ec2.internal systemd[1]: crio.service: main process exited, code=exited, status=1/FAILURE
Sep 28 20:06:03 ip-172-18-8-38.ec2.internal systemd[1]: Failed to start Open Container Initiative Daemon.
Sep 28 20:06:03 ip-172-18-8-38.ec2.internal systemd[1]: Unit crio.service entered failed state.
Sep 28 20:06:03 ip-172-18-8-38.ec2.internal systemd[1]: crio.service failed.
```

**Comment by rhatdan** [comment]
@mrunalp removed your patch and repushed.

**Comment by mrunalp** [comment]
@rhatdan thanks! I am trying to verify that in a AWS VM manually.

**Comment by rhatdan** [comment]
@mrunalp Seems like e2e_rhel is hung,  I think this is ready to go.  Should we just merge?

**Comment by stevekuznetsov** [comment]
In the case CRI-O doesn't come up or something else critical fails .. e2e takes 4+ hours for all tests to time out and fail

**Comment by mrunalp** [comment]
okay, the e2e rhel in the last run only had the guestbook failure so this is good. @rhatdan Did we ge the other tests to pass after you changed the error to warning?

**Comment by rhatdan** [comment]
@mrunalp Yes.  I believe so.

**Comment by rhatdan** [comment]
Since this now passes the tests other then the known failure.  I am going to merge.

**Comment by rhatdan** [comment]
@mrunalp Since this is merged, are we ready for RC3?

**Comment by mrunalp** [comment]
Looks like this broke the build. See #963.

**Pull request merged**